### PR TITLE
Fixed ReflectionMethod::isAbstract() in adapter

### PR DESCRIPTION
**Pull request by kukulich**
See https://3v4l.org/sdhSP

### CODE REVIEW

File: src/Reflection/Adapter/ReflectionMethod.php [hunk 1]
```diff
@@ -273,7 +273,7 @@ public function isProtected()
      */
     public function isAbstract()
     {
-        return $this->betterReflectionMethod->isAbstract();
+        return $this->betterReflectionMethod->isAbstract() || $this->betterReflectionMethod->getDeclaringClass()->isInterface();
```
Comment (asgrim) [inline src/Reflection/Adapter/ReflectionMethod.php:276]: @kukulich I think this should probable be fixed in `\Roave\BetterReflection\Reflection\ReflectionClass::isAbstract` too, perhaps?

**Comment by asgrim** [comment]
Oops, merged the wrong MR :roll_eyes: @kukulich see comment above :grin:

File: src/Reflection/Adapter/ReflectionMethod.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/Adapter/ReflectionMethod.php:276]: It looks the current version is right: https://3v4l.org/EBjjL

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionMethod.php:276]: wow, PHP is confusing 🤣

Comment (asgrim) [reply src/Reflection/Adapter/ReflectionMethod.php:276]: Oh, I made a mistake, I meant the BR ReflectionMethod @kukulich 🙄 sorry

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionMethod.php:276]: Reported in #596

Comment (kukulich) [reply src/Reflection/Adapter/ReflectionMethod.php:276]: @asgrim I'm not sure. BR is "better" so it should maybe report only really `abstract` methods...

But there's `ReflectionMethod::getModifiers()` based also on `isAbstract()`...

You tell me how I should fix it :D

Comment (ondrejmirtes) [reply src/Reflection/Adapter/ReflectionMethod.php:276]: I wasn't kidding when I said that PHPStan is an integration test suite for BetterReflection :D https://twitter.com/OndrejMirtes/status/1261621042176548865

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionMethod.php:276]: > You tell me how I should fix it :D

Oooooooooof

Guess we'll need more tests, but if you have a bug or scenario in mind, report a new issue referencing this thread.

The test suite and code are in good shape enough to "outsource" all of this work to the twitters :-)

**Pull request merged**